### PR TITLE
Resolves errant privilege escalation behavior

### PR DESCRIPTION
**Pull request by eddie-knight**

### CODE REVIEW

**Comment by eddie-knight** [comment]
I included the updated `examples/cucumber_output` here because those have changed quite a bit

File: service_packs/kubernetes/container_registry_access/container_registry_access.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func (s *scenarioState) aKubernetesClusterIsDeployed() error {
 func (s *scenarioState) iAmAuthorisedToPullFromAContainerRegistry() error {
 	pod, podAudit, err := cra.SetupContainerAccessProbePod(config.Vars.AuthorisedContainerRegistry, s.probe)
 
-	err = kubernetes.ProcessPodCreationResult(s.probe, &s.podState, pod, kubernetes.PSPContainerAllowedImages, err)
```
Comment (eddie-knight) [inline service_packs/kubernetes/container_registry_access/container_registry_access.go:42]: sortof a refactor... noticed this arg wasn't being used anymore.

File: service_packs/kubernetes/kube.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func GetKubeInstance() *Kube {
 		instance.cspErrorToProbrCreationError["azurepolicy-container-no-privilege"] = PSPNoPrivilege
 		instance.cspErrorToProbrCreationError["securityContext.privileged: Invalid value: true"] = PSPNoPrivilege
 
-		instance.cspErrorToProbrCreationError["azurepolicy-psp-container-no-privilege-escalation"] = PSPNoPrivilegeEscalation
+		instance.cspErrorToProbrCreationError["azurepolicy-psp-container-no-privilege-esc"] = PSPNoPrivilegeEscalation
```
Comment (eddie-knight) [inline service_packs/kubernetes/kube.go:116]: name didn't match, this didn't work by itself but combined with the new yaml file it fixed the unexpected behavior

File: service_packs/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -63,22 +63,22 @@ func (s *scenarioState) setup() {
 
 // ProcessPodCreationResult is a convenience function to process the result of a pod creation attempt.
 // It records state information on the supplied state structure.
-func ProcessPodCreationResult(probe *summary.Probe, s *PodState, pd *apiv1.Pod, expected PodCreationErrorReason, err error) error {
+func ProcessPodCreationResult(state *PodState, pd *apiv1.Pod, expected PodCreationErrorReason, err error) error {
```
Comment (eddie-knight) [inline service_packs/kubernetes/kubernetes.go:66]: part of the aforementioned refactor... decided to lengthen the name here.

File: service_packs/kubernetes/pod_security_policy/pod_security_policy.go [hunk 4]
```diff
@@ -304,18 +305,18 @@ func (s *scenarioState) iShouldNotBeAbleToPerformACommandThatProvidesAccessToThe
...
 func (s *scenarioState) privilegedEscalationIsMarkedForTheKubernetesDeployment(privilegedEscalationRequested string) error {
 
-	var pa bool
-	if privilegedEscalationRequested == "True" {
-		pa = true
-	} else {
-		pa = false
+	allowPrivilegeEscalation := "true"
+	if strings.ToLower(privilegedEscalationRequested) != "true" {
+		allowPrivilegeEscalation = "false"
```
Comment (eddie-knight) [inline service_packs/kubernetes/pod_security_policy/pod_security_policy.go:310]: Any values that are not True/true should resolve to "false" for the purposes of this step.

**Review by simonwoolhouse — APPROVED** [review]

**Pull request merged**